solutions/cses-1676.mdx

```diff
@@ -2,17 +2,25 @@
 id: cses-1676
 source: CSES
 title: Road Construction
-author: Oscar Garries, Sofia Yang
+author: Oscar Garries, Sofia Yang, George Pong
 ---
 
-**Time Complexity:** $\mathcal{O}(N\log N)$
+## Explanation
 
-<LanguageSection>
+We can represent the problem as an undirected graph, with each of the $n$ cities being a node and each of the $m$ roads being an edge. We then have two subproblems we need to consider:
 
-<CPPSection>
+Our first subproblem is counting the number of components each day. By the definition of a component given in the problem, a component is a connected component. We can count them using Disjoint Set Union (DSU), where each tree in the structure represents a connected component. We start off with $n$ cities and no roads between them, and thus $n$ connected components. For every road that is built, we union the two cities the road connects. If the union is successful, then two different connected components have been joined into one, and thus the number of connected components decreases by 1.
```

Comment:
> Our first subproblem is counting the number of components each day. By the definition of a component given in the problem, a component is a connected component. 

idk maybe just `Our first subproblem is counting the number of connected components each day`?